# Instalando pacotes

In [2]:
# Install requirements
!pip install --upgrade pip
!pip install pandas
!pip install matplotlib
!pip install numpy

/usr/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory
/usr/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory
/usr/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory
/usr/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory


# Importando dependências e a base que será utilizada no projeto

In [3]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

credit_risk = pd.read_csv('german_credit_risk_mineração_BI.csv')

# Initial Database
credit_risk.head()

Unnamed: 0  Age     Sex  Job Housing Saving accounts Checking account  \
0           0   67    male    2     own             NaN           little   
1           1   22  female    2     own          little         moderate   
2           2   49    male    1     own          little              NaN   
3           3   45    male    2    free          little           little   
4           4   53    male    2    free          little           little   

   Credit amount  Duration              Purpose  Risk  
0           1169         6             radio/TV  good  
1           5951        48             radio/TV   bad  
2           2096        12            education  good  
3           7882        42  furniture/equipment  good  
4           4870        24                  car   bad

# Análise Exploratória de Dados

In [4]:
# View all Dimensions
dimensions = credit_risk.shape
print(f'Dimensions: {dimensions}')

Dimensions: (1000, 11)


In [5]:
# Some Statistics from credit risk
credit_risk.describe()

Unnamed: 0          Age          Job  Credit amount     Duration
count  1000.000000  1000.000000  1000.000000    1000.000000  1000.000000
mean    499.500000    35.546000     1.904000    3271.258000    20.903000
std     288.819436    11.375469     0.653614    2822.736876    12.058814
min       0.000000    19.000000     0.000000     250.000000     4.000000
25%     249.750000    27.000000     2.000000    1365.500000    12.000000
50%     499.500000    33.000000     2.000000    2319.500000    18.000000
75%     749.250000    42.000000     2.000000    3972.250000    24.000000
max     999.000000    75.000000     3.000000   18424.000000    72.000000

> Observando já com o `describe` consigo entender alguns números:

. As idades variam em média cerca de 11.4 anos em relação à média que é de 35.5 anos

. Os valores de crédito tem uma dispersão bem alta

. A duração dos empréstimos varia em média cerca de 12 meses

# Perguntas de Estatística Descritiva e Pré-processamento


### 1. Qual é a média, mediana e desvio padrão da idade dos clientes?

In [7]:
# Describe only for Age column
describe_age_clients = credit_risk['Age'].describe()
describe_age_clients

count    1000.000000
mean       35.546000
std        11.375469
min        19.000000
25%        27.000000
50%        33.000000
75%        42.000000
max        75.000000
Name: Age, dtype: float64

In [11]:
# Separating only Age statistics with mean, median and std
mean_age = credit_risk['Age'].mean()
median_age = credit_risk['Age'].median()
std_age = credit_risk['Age'].std()

# Transforming to DataFrame
statistics_age =  np.array([mean_age, median_age, std_age])
df_age = pd.DataFrame(statistics_age, index=['mean', 'median', 'std'], columns=['Age'])
df_age

Age
mean    35.546000
median  33.000000
std     11.375469

### Resposta:

. **Média**: 35,5

. **Mediana**: 33

. **Desvio Padrão**: 11,4

### 2. Qual é a distribuição de valores nulos nas colunas `Saving accounts` e `Checking account`?

### 3. Qual é a proporção de clientes por categoria de `Sex` e `Housing`?

### 4. Quais são os valores mínimos, máximos e quartis da coluna `Credit amount`? Existem outliers?

### 5. Como está distribuído o tempo de `Duration` dos empréstimos (média, mediana, outliers)?

### 6. Quais são as categorias de `Purpose` mais frequentes e as suas porcentagens?

### 7. Qual é a proporção de clientes com risco `good` e `bad`?

### 8. Existe diferença na média do `Credit amount` entre clientes com diferentes categorias de `Saving accounts`?

### 9. A idade média difere entre clientes classificados como `good` e `bad` risco?

### 10. Como se relacionam `Job` e `Housing`? Qual categoria de moradia é mais comum em cada tipo de trabalho?